### ISSUE TITLE
MacOS Version 3.8.1 crash on both M1 and Intel MacBook Pro

### ISSUE THREAD

**Issue by playniuniu** [RocketChat/Rocket.Chat.Electron#2373]
**Describe the bug**

MacOS Version 3.8.1 crash on both M1 and Intel MacBook Pro

What operating system and which version?

MacOS 12.3.1

Which version of Rocket.Chat (Server)?

Unknown

Which version of Rocket.Chat.Electron (Electron/Desktop)?

3.8.1

Is there any setting relevant changed?

No

**To Reproduce**
Download Rocket.Chat 3.8.1 From AppStore
Open and then crash every time.
Both M1 and Intel MacBook Pro have the same problem.
Version 3.8.0 works fine and not crash.

**Comment by mlemnian** [comment]
same on my intel based MacBook Pro with MacOS 10.15.7

**Comment by gufertum** [comment]
+1

**Comment by sbnmrk** [comment]
crash log added, so this issue can be searched for ...

```
-------------------------------------

Process:               Rocket.Chat [3152]
Path:                  /Applications/Rocket.Chat.app/Contents/MacOS/Rocket.Chat
Identifier:            chat.rocket
Version:               3.8.1 (83)
App Item ID:           1086818840
App External ID:       848006747
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]

OS Version:            macOS 12.3.1 (21E258)
Report Version:        12
Bridge OS Version:     6.4 (19P4243)
Anonymous UUID:        575C6C72-9C07-19CC-BD9F-14687325729D


System Integrity Protection: enabled

Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000002, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 5 Trace/BPT trap: 5
Terminating Process:   exc handler [3152]

Thread 0 Crashed:: CrBrowserMain Dispatch queue: com.apple.main-thread
0   Electron Framework            	       0x122dbc282 node::AsyncResource::get_async_id() const + 8249314
1   Electron Framework            	       0x12222e0ea v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6509914
2   Electron Framework            	       0x12222b86e v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6499550
3   Electron Framework            	       0x12222a4d3 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6494531
4   Electron Framework            	       0x1222351a7 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6538775
5   Electron Framework            	       0x1222364a7 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6543639
6   Electron Framework            	       0x1220d22fa v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5085034
7   Electron Framework            	       0x1233590e0 node::AsyncResource::get_async_id() const + 14134848
8   Electron Framework            	       0x12313995d node::AsyncResource::get_async_id() const + 11908797
9   Electron Framework            	       0x12335e09d node::AsyncResource::get_async_id() const + 14155261
10  Electron Framework            	       0x12335f631 node::AsyncResource::get_async_id() const + 14160785
11  Electron Framework            	       0x123135c53 node::AsyncResource::get_async_id() const + 11893171
12  Electron Framework            	       0x123159bcc node::AsyncResource::get_async_id() const + 12040492
13  Electron Framework            	       0x122df936e node::AsyncResource::get_async_id() const + 8499406
14  Electron Framework            	       0x122e17169 node::AsyncResource::get_async_id() const + 8621769
15  Electron Framework            	       0x122e55905 node::AsyncResource::get_async_id() const + 8877669
16  Electron Framework            	       0x122e54a72 node::AsyncResource::get_async_id() const + 8873938
17  Electron Framework            	       0x122e5503f node::AsyncResource::get_async_id() const + 8875423
18  CoreFoundation                	    0x7ff810117aeb __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
19  CoreFoundation                	    0x7ff810117a53 __CFRunLoopDoSource0 + 180
20  CoreFoundation                	    0x7ff8101177cd __CFRunLoopDoSources0 + 242
21  CoreFoundation                	    0x7ff8101161e8 __CFRunLoopRun + 892
22  CoreFoundation                	    0x7ff8101157ac CFRunLoopRunSpecific + 562
23  HIToolbox                     	    0x7ff818d9cce6 RunCurrentEventLoopInMode + 292
24  HIToolbox                     	    0x7ff818d9ca4a ReceiveNextEventCommon + 594
25  HIToolbox                     	    0x7ff818d9c7e5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
26  AppKit                        	    0x7ff812b3c53d _DPSNextEvent + 927
27  AppKit                        	    0x7ff812b3abfa -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1394
28  AppKit                        	    0x7ff812b2d2a9 -[NSApplication run] + 586
29  Electron Framework            	       0x122e56216 node::AsyncResource::get_async_id() const + 8879990
30  Electron Framework            	       0x122e54ae2 node::AsyncResource::get_async_id() const + 8874050
31  Electron Framework            	       0x122e17afa node::AsyncResource::get_async_id() const + 8624218
32  Electron Framework            	       0x122de1801 node::AsyncResource::get_async_id() const + 8402273
33  Electron Framework            	       0x1220eac44 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5185716
34  Electron Framework            	       0x1220ec582 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5192178
35  Electron Framework            	       0x1220e8350 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5175232
36  Electron Framework            	       0x12029f2e5 v8::internal::ExternalReference::fast_c_call_caller_pc_address(v8::internal::Isolate*) + 1957
37  Electron Framework            	       0x1202a04b3 v8::internal::ExternalReference::fast_c_call_caller_pc_address(v8::internal::Isolate*) + 6515
38  Electron Framework            	       0x12029ff31 v8::internal::ExternalReference::fast_c_call_caller_pc_address(v8::internal::Isolate*) + 5105
39  Electron Framework            	       0x12029e4ed v8::internal::compiler::BasicBlock::set_loop_header(v8::internal::compiler::BasicBlock*) + 12173
40  Electron Framework            	       0x12029ee23 v8::internal::ExternalReference::fast_c_call_caller_pc_address(v8::internal::Isolate*) + 739
41  Electron Framework            	       0x11ffd4ad8 ElectronMain + 152
42  dyld                          	       0x1100e551e start + 462
```

**Comment by airbenich** [comment]
Same on my M1 machine. Here is the console log:

`
user@MacBook-Pro/ % ./Applications/Rocket.Chat.app/Contents/MacOS/Rocket.Chat 
[87737:0404/100605.942902:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100605.957566:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.119771:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.136614:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.198295:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.254487:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.277996:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.340183:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.407615:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.408548:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.475156:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.541941:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.571073:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.634663:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.692933:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.752381:ERROR:network_service_instance_impl.cc(975)] Network service crashed, restarting service.
[87737:0404/100606.788510:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.848845:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=4
[87737:0404/100606.848867:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.
zsh: trace trap  ./Applications/Rocket.Chat.app/Contents/MacOS/Rocket.Chat
user@MacBook-Pro/ % 

`

**Comment by florian-hofmann-1977** [comment]
Same here on M1 MBA

**Comment by airbenich** [comment]
Duplicate to: https://github.com/RocketChat/Rocket.Chat.Electron/issues/2372
@playniuniu Maybe you can close this issue.

**Comment by kenroyandfamily** [comment]
Rocket.Chat 3.8.1 crashes for me too.   iMac (Retina 5K, 27-inch 2020) running macOS 12.1
I downloaded this repo and 3.8.2 is running without issue.

**Comment by melligeorgiou** [comment]
Same here (intel macbook pro) - crashed after auto-update from mac app store.

**Comment by rmdrabach** [comment]
Maybe would be the case for an expedite review...

**Comment by RashidRysaev** [comment]
> The issue is only with the Mac Apple Store executable. The same version installed without Apple Store will work as usually.

Thank you, you made my day.

**Comment by playniuniu** [comment]
Fixed with version 3.8.2
